compositor_pipeline/src/audio_mixer/prepare_inputs.rs

```diff
@@ -71,8 +76,8 @@ fn frame_input_samples(
     let time_to_sample_count = |duration: Duration| {
         let sample_count = duration.as_secs_f64() * sample_rate as f64;
         // If value is close to the integer then round it, otherwise fallback to standard
-        // integer division behavior.
-        if (sample_count - sample_count.round()).abs() < max_error.as_secs_f64() {
+        // integer division behavior. Close is defined as 1% of a sample (the same as max_error).
+        if (sample_count - sample_count.round()).abs() < 0.01 {
```

Comment:
max error represents the duration that reflects 1% of a sample and in this condition, we compare sample count not duration. 

max_error should be about 1e-8 seconds and sample duration is about 1e-5 seconds so the old implementation was only allowing for errors of the order 1e-13 seconds